src/main/java/com/bettercloud/vault/api/Leases.java

```diff
@@ -45,14 +51,22 @@ public VaultResponse revoke(final String leaseId) throws VaultException {
         int retryCount = 0;
         while (true) {
             try {
-                final RestResponse restResponse = new Rest()//NOPMD
+                final RestResponse restResponse;
+                final Rest rest = new Rest()//NOPMD
                         .url(config.getAddress() + "/v1/sys/revoke/" + leaseId)
                         .header("X-Vault-Token", config.getToken())
                         .connectTimeoutSeconds(config.getOpenTimeout())
                         .readTimeoutSeconds(config.getReadTimeout())
                         .sslVerification(config.getSslConfig().isVerify())
-                        .sslContext(config.getSslConfig().getSslContext())
-                        .put();
+                        .sslContext(config.getSslConfig().getSslContext());
+
+                if (this.nameSpace != null && !this.nameSpace.isEmpty()) {
```

Comment:
minor suggestion, but if you added an .optionalHeader() to the Rest class you could move the != null and !isEmpty() checks into that, and you would not need to add so many lines to each use of the namespace?

Reply:
Another note: worth checking this works with sub-namespaces that have a / in them - e.g. foo/bar - as previously others had reported this required the .header() to not URL-encode the value, otherwise this isn't sent as foo/bar - see #148 for example

Reply:
Other than those points, it looks good to me. (have not been able to test myself yet, though)

Reply:
We're going to turn off that encoding, as the %2F is still present. I also like your optionalHeader idea. I can get this done shortly.

Reply:
Looks good, thanks @jarrodcodes !